subgraphs/hop-protocol/protocols/hop-protocol/src/mainnet/bridge.ts

```diff
@@ -0,0 +1,255 @@
+import { SDK } from "../../../../src/sdk/protocols/bridge";
+import { TokenPricer } from "../../../../src/sdk/protocols/config";
+import {
+  TokenInitializer,
+  TokenParams,
+} from "../../../../src/sdk/protocols/bridge/tokens";
+import {
+  BridgePermissionType,
+  CrosschainTokenType,
+  BridgePoolType,
+} from "../../../../src/sdk/protocols/bridge/enums";
+import { BridgeConfig } from "../../../../src/sdk/protocols/bridge/config";
+import { Versions } from "../../../../src/versions";
+import { NetworkConfigs } from "../../../../configurations/configure";
+import {
+  Address,
+  BigDecimal,
+  BigInt,
+  Bytes,
+  dataSource,
+  log,
+} from "@graphprotocol/graph-ts";
+import {
+  TransferSentToL2,
+  BonderAdded,
+} from "../../../../generated/HopL1Bridge/L1_Bridge";
+
+import { Token } from "../../../../generated/schema";
+import { getUsdPricePerToken, getUsdPrice } from "../../../../src/prices/index";
+import { bigIntToBigDecimal } from "../../../../src/sdk/util/numbers";
+import {
+  ARBITRUM_L1_SIGNATURE,
+  MESSENGER_EVENT_SIGNATURES,
+  OPTIMISM_L1_SIGNATURE,
+  XDAI_L1_SIGNATURE,
+} from "../../../../src/common/constants";
+
+class Pricer implements TokenPricer {
+  getTokenPrice(token: Token): BigDecimal {
+    const price = getUsdPricePerToken(Address.fromBytes(token.id));
+    return price.usdPrice;
+  }
+
+  getAmountValueUSD(token: Token, amount: BigInt): BigDecimal {
+    const _amount = bigIntToBigDecimal(amount, token.decimals);
+    return getUsdPrice(Address.fromBytes(token.id), _amount);
+  }
+}
+
+class TokenInit implements TokenInitializer {
+  getTokenParams(address: Address): TokenParams {
+    const tokenConfig = NetworkConfigs.getTokenDetails(address.toHex());
+    const name = tokenConfig[1];
+    const symbol = tokenConfig[0];
+    const decimals = BigInt.fromString(tokenConfig[2]).toI32();
+    return { name, symbol, decimals };
+  }
+}
+
+const conf = new BridgeConfig(
+  "0x03D7f750777eC48d39D080b020D83Eb2CB4e3547",
+  "HOP-"
+    .concat(dataSource.network().toUpperCase().replace("-", "_"))
+    .concat("-BRIDGE"),
+  "hop-".concat(dataSource.network().replace("-", "_")).concat("-bridge"),
+  BridgePermissionType.PERMISSIONLESS,
+  Versions
+);
+
+export function handleBonderAdded(event: BonderAdded): void {
+  if (NetworkConfigs.getBridgeList().includes(event.address.toHexString())) {
+    log.warning("bridgeAddress: {}", [event.address.toHexString()]);
+
+    const sdk = SDK.initializeFromEvent(
+      conf,
+      new Pricer(),
+      new TokenInit(),
+      event
+    );
+    sdk.Accounts.loadAccount(event.params.newBonder);
+  }
+}
+
+export function handleTransferSentToL2(event: TransferSentToL2): void {
+  if (event.params.chainId.toString() == "42170") return;
```

Comment:
This is a chain Arbitrum Nova. It broke the subgraph because that network isn't tracked. 
It isn't officially supported.

Reply:
Left this. The mapping logic breaks if chainId is unrecognised.

Reply:
Can we add this chain to the `Network` enum in the schema and then support transfers to that chain? Without it we are missing TVL exported there and netVolume.

As a side note: If we were to leave this in, It would be good to add a comment in the code explaining this early return, and to make the `42170` portion a constant

Reply:
Okay. Looks like they just added Arbitrum Nova support. ETH and MAGIC are the tokens supported for now.